pinot-core/src/test/java/org/apache/pinot/queries/ExplainPlanQueriesTest.java

```diff
@@ -0,0 +1,510 @@
+            + "noIndexCol3)))", 2, 1});
+    result.add(new Object[]{"PROJECT(noIndexCol3, noIndexCol2, noIndexCol1)", 3, 2});
+    result.add(new Object[]{"FILTER_MATCH_ENTIRE_SEGMENT(docs:1)", 4, 3});
+    check(query, new ResultTable(DATA_SCHEMA, result));
+  }
+
+  @Test
+  public void testSelectAggregateUsingFilterGroupBy() {
+    String query =
+        "EXPLAIN PLAN FOR SELECT noIndexCol2, sum(add(noIndexCol1, noIndexCol2)), min(noIndexCol3) FROM testTable "
+            + "WHERE noIndexCol1 < 100 GROUP BY noIndexCol2";
+    List<Object[]> result = new ArrayList<>();
+    result.add(new Object[]{"BROKER_REDUCE(limit:10)", 0, -1});
+    result.add(new Object[]{"COMBINE_GROUPBY_ORDERBY", 1, 0});
+    result.add(new Object[]{
+        "AGGREGATE_GROUPBY_ORDERBY(groupKeys:noIndexCol2, aggregations:sum(add(noIndexCol1,noIndexCol2)),"
+            + " min(noIndexCol3))", 2, 1});
+    result.add(new Object[]{"PROJECT(noIndexCol3, noIndexCol2, noIndexCol1)", 3, 2});
+    result.add(new Object[]{"FILTER_MATCH_ENTIRE_SEGMENT(docs:1)", 4, 3});
+    check(query, new ResultTable(DATA_SCHEMA, result));
+  }
+
+  @Test
+  public void testSelectAggregateUsingFilterIndex() {
+    String query =
+        "EXPLAIN PLAN FOR SELECT count(*), max(noIndexCol1), sum(noIndexCol2), avg(noIndexCol3) FROM testTable WHERE "
+            + "invertedIndexCol1 = 1 AND col1 = 20";
+    List<Object[]> result = new ArrayList<>();
+    result.add(new Object[]{"BROKER_REDUCE(limit:10)", 0, -1});
+    result.add(new Object[]{"COMBINE_AGGREGATE", 1, 0});
+    result.add(
+        new Object[]{"AGGREGATE(aggregations:count(*), max(noIndexCol1), sum(noIndexCol2), avg(noIndexCol3))", 2, 1});
+    result.add(new Object[]{"TRANSFORM_PASSTHROUGH(transformFuncs:noIndexCol1, noIndexCol2, noIndexCol3)", 3, 2});
+    result.add(new Object[]{"PROJECT(noIndexCol3, noIndexCol2, noIndexCol1)", 4, 3});
+    result.add(new Object[]{"FILTER_EMPTY", 5, 4});
+    check(query, new ResultTable(DATA_SCHEMA, result));
+  }
+
+  @Test
+  public void testSelectAggregateUsingFilterIndexGroupBy() {
+    String query = "EXPLAIN PLAN FOR SELECT noIndexCol1, max(noIndexCol2), min(noIndexCol3) FROM testTable WHERE "
+        + "invertedIndexCol2 = 10 GROUP BY noIndexCol1";
+    List<Object[]> result = new ArrayList<>();
+    result.add(new Object[]{"BROKER_REDUCE(limit:10)", 0, -1});
+    result.add(new Object[]{"COMBINE_GROUPBY_ORDERBY", 1, 0});
+    result.add(new Object[]{
+        "AGGREGATE_GROUPBY_ORDERBY(groupKeys:noIndexCol1, aggregations:max(noIndexCol2), min(noIndexCol3)" + ")", 2,
+        1});
+    result.add(new Object[]{"TRANSFORM_PASSTHROUGH(transformFuncs:noIndexCol1, noIndexCol2, noIndexCol3)", 3, 2});
+    result.add(new Object[]{"PROJECT(noIndexCol3, noIndexCol2, noIndexCol1)", 4, 3});
+    result.add(new Object[]{"FILTER_EMPTY", 5, 4});
+    check(query, new ResultTable(DATA_SCHEMA, result));
+  }
+
+  @Test
+  public void testSelectAggregateUsingFilterIndexGroupByHaving() {
```

Comment:
Please add following tests to improve coverage of this functionality

- Transform in filter. Example - `AND arrayLength(col) > 2 " + "AND invertedIndexCol1 * 5 < 1000`
- Transform in select list and GROUP BY and ORDER BY. The plan should show the actual transformation happening. Right now the only transform we show is the PASS_THROUGH transform on identifiers
- Transform in select list inside aggregations. You have this but transform part is skipped from the plan which is not correct

Reply:
Transform in:
- filter: see `testSelectColumnsUsingFilterIndex`.
- select: see `testSelectAggregate`
- group by and order by: see `testSelectAggregateUsingFilterIndexGroupByOrderBy`

Reply:
Thanks for adding remaining tests